packages/common/src/location/location.ts

```diff
@@ -301,9 +301,15 @@ export function createLocation() {
 }
 
 function _stripBasePath(basePath: string, url: string): string {
-  return basePath && new RegExp(`^${basePath}([/;?#]|$)`).test(url) ?
-      url.substring(basePath.length) :
-      url;
+  if (!basePath || !url.startsWith(basePath)) {
+    return url;
+  }
+  if (url.length === basePath.length) {
+    return '';
+  }
+  return ['/', ';', '?', '#'].includes(url[basePath.length]) ?
+    url.substring(basePath.length) :
+    url;
```

Comment:
This code is a bit confusing, especially `array.includes`. 

What do you think of this instead?
```suggestion
  const strippedBasePath = url.substring(basePath.length);
  const nonSegmentTerminator = new RegExp(`[^/;?#]`);
  // Ensure that the stripped path does not start with a character that's not a segment terminator.
  // This is because the URL does not technically match the base path.
  // i.e. if base path is a/b/c and the URL is a/b/cat.
  return nonSegmentTerminator.test(strippedBasePath[0]) ? url : strippedBasePath;
```